.github/workflows/test.yml

```diff
@@ -0,0 +1,37 @@
+name: Pull request
+
+on:
+  pull_request:
+    paths:
+      - '/packages/**'
```

Comment:
I think to be safe the tests would have to run if either anything in `/package` has changed OR the tests themselves (in `/tests`) OR in `/app` - as the visual regression tests rely on examples within `/app` to check the rendering?

...but we could skip the tests if a PR doesn’t touch any of those (eg a change to docs)?

We’d also have to change the branch protection rules to turn off the requirements for the tests to pass (which might defeat the point?) - unless there’s a way for the job to succeed without actually running the tests, if none of files in the paths have been changed? 

```suggestion
      - '/packages/**'
      - '/tests/**'
      - '/app/**'
```

Reply:
Good point, updated. I can’t see the branch protection rules, but am assuming only test that are run need to pass; if the visual regression test isn’t run then it won’t fail, no?

Reply:
@paulrobertlloyd if it’s not run then it'll show as pending and won't let you merge the pull request (unless you let admins override it). I had a quick look at the `paths` documentation in GitHub but couldn’t see an obvious way around it.

Reply:
Doesn’t seem possible:

<img width="777" alt="Screenshot 2024-11-28 at 11 34 25" src="https://github.com/user-attachments/assets/dc641cd1-f584-44be-a190-4ce2c5432829">

Reply:
Ah, bugger. Looks like the path scoping needs to happen in the branch protection rules, not the workflow.

I wonder if it might be easier to update the visual regression test script to pass if no files have changed instead?